packages/sync-actions/src/customer-actions.js

```diff
@@ -126,3 +141,61 @@ export function actionsMapShippingAddresses(diff, oldObj, newObj) {
 
   return handler(diff, oldObj, newObj)
 }
+
+export function actionsMapAuthenticationModes(diff, oldObj, newObj) {
+  // eslint-disable-next-line no-use-before-define
+  return buildAuthenticationModeActions({
+    actions: authenticationModeActionsList,
+    diff,
+    oldObj,
+    newObj,
+  })
+}
+
+function buildAuthenticationModeActions({ actions, diff, oldObj, newObj }) {
+  return actions
+    .map((item) => {
+      const key = item.key
+      const value = item.value || item.key
+      const delta = diff[key]
+      const before = oldObj[key]
+      const now = newObj[key]
+      const isNotDefinedBefore = isEmptyValue(oldObj[key])
+      const isNotDefinedNow = isEmptyValue(newObj[key])
+
+      if (!delta) return undefined
+
+      if (isNotDefinedNow && isNotDefinedBefore) return undefined
+
+      if (newObj.authenticationMode === 'Password' && !newObj.password)
+        throw new Error(
+          'Cannot set to Password authentication mode without password'
+        )
+
+      if (!isNotDefinedNow && isNotDefinedBefore) {
+        // no value previously set
+        if (newObj.authenticationMode === 'ExternalAuth')
+          return { action: item.action, authMode: now }
+        return { action: item.action, authMode: now, [value]: newObj.password }
```

Comment:
Codecov created this assumtion whats if the user input isnt trusted, ex: if user provided  authMode as 'xyz'
this will be invalid action created, just random thought, maybe 
```
if (newObj.authenticationMode === 'Password')
  return { action: item.action, authMode: 'Password', [value]: newObj.password }
```
  wdyt?

Reply:
as always, good catch, thanks! i added a case for this and updated a previous one. so invalid strings (e.g. 'xyz')  and empty strings will throw an error. and if key is not present, it will not create an action. wdyt?